haas/client/node.py

```diff
@@ -0,0 +1,33 @@
+from haas.client.base import *
+from haas.client.client_errors import *
+
+
+class Node(ClientBase):
+    """ Consists of calls to query and manipulate node related
+    objects and relations """
+
+    def list(self, is_free):
+        """ List all nodes that HIL manages """
+        url = self.object_url('node', is_free)
+        q = requests.get(url, headers={"Authorization": "Basic %s" % self.auth})
```

Comment:
If we use the HTTPClient implementation, we don't need to do basic auth here.

Also, requests allows you to [pass an `auth` parameter](http://docs.python-requests.org/en/master/user/authentication/) so you don't have an manually insert a header line.